server/command.go

```diff
@@ -33,13 +33,18 @@ func (p *Plugin) createCommand(syncLinkedChannels bool) *model.Command {
 	}
 }
 
-func (p *Plugin) cmdError(userID string, channelID string, detailedError string) (*model.CommandResponse, *model.AppError) {
-	p.API.SendEphemeralPost(userID, &model.Post{
-		Message:   detailedError,
-		UserId:    p.userID,
-		ChannelId: channelID,
-	})
-	return &model.CommandResponse{}, nil
+func (p *Plugin) cmdSuccess(text string) (*model.CommandResponse, *model.AppError) {
+	return &model.CommandResponse{
+		ResponseType: model.CommandResponseTypeEphemeral,
+		Text:         text,
+	}, nil
+}
+
+func (p *Plugin) cmdError(detailedError string) (*model.CommandResponse, *model.AppError) {
```

Comment:
In the happy path, we don't need to send our own ephemeral messages but can have the server do that -- plus, the unit tests are simpler when we can simply test the returned `model.CommandResponse` in most cases. (In others, where we still need ephemeral messages, I do go to the effort of monitoring the websocket directly.)

Reply:
I ended up reverting this part to continue to send ephemeral messages from the bot. I realized we can't reply with a non-system user, so the experience changed. I also discovered that we needed some extra helper functionality to monitor for websocket events to ensure correct behaviour.